src/clr-core/common/utils/register.ts

```diff
@@ -1,19 +1,32 @@
 /*
- * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
 
+import and from 'ramda/es/and';
 import curryN from 'ramda/es/curryN';
-import __ from './__';
-import { elementExists } from './exists';
+import isNil from 'ramda/es/isNil';
+import { elementExists, existsInWindow } from './exists';
 
-const addElementToRegistry = curryN(3, (tagName: string, elementClass: any, registry = window.customElements) => {
-  if (elementExists(tagName)) {
-    console.warn(`${tagName} has already been registered`);
-  } else {
-    registry.define(tagName, elementClass);
+const addElementToRegistry = curryN(
+  3,
+  (tagName: string, elementClass: any, registry: { define: (a: string, b: any) => {} }) => {
+    if (elementExists(tagName)) {
+      console.warn(`${tagName} has already been registered`);
+    } else {
+      registry.define(tagName, elementClass);
+    }
   }
-});
+);
 
-export const registerElementSafely = addElementToRegistry(__, __, window.customElements);
+export function registerElementSafely(tagName: string, elementClass: any) {
+  const guard = and(!isNil(window), existsInWindow(['customElements']));
```

Comment:
nit: Could this be extracted to a function like `isBrowser()`?

Is there an advantage of `and` vs `&&` if its not used as a curried func?

Reply:
Re: and. Nope. I can change that. 

Re: shared function. Yup. Will do.

Reply:
Re: and. Nope. I can change that. 

Re: shared function. Yup. Will do.